js/framework/modules/view/document.js

```diff
@@ -266,7 +266,7 @@ var Document = new Module.Class({
     _get_webview: function () {
         let webview = this._create_webview();
 
-        webview.renderer.enable_scroll_manager = true;
+        webview.renderer.enable_scroll_manager = this.toc.visible;
```

Comment:
Out of curiosity, is this a fix for a pre-existing bug or is it made necessary by this change? It seems like it can improve performance on apps that don't use the table of contents widget, since I think that scroll-manager script makes things slow.

Reply:
Exactly, with this change, current legacy apps which doesn't show the side TOC, will disable the scroll manager. This change was actually necessary anyway besides the fine side effect, since without it, absolutely all apps (legacy or not) will enable it.